conans/client/generators/cmake_common.py

```diff
@@ -323,8 +323,19 @@ def generate_targets_section(dependencies):
 endmacro()
 
 macro(conan_error_compiler_version)
-    message(FATAL_ERROR "Incorrect '${CONAN_COMPILER}' version 'compiler.version=${CONAN_COMPILER_VERSION}'"
-                        " is not the one detected by CMake: '${CMAKE_CXX_COMPILER_ID}=" ${VERSION_MAJOR}.${VERSION_MINOR}')
+    if(NOT "${CONAN_COMPILER}" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
+        message(FATAL_ERROR "Detected a mismatch between your conan profile settings and the compiler CMake has detected:\n"
```

Comment:
I think this is not necessary, the file already has a check for compiler mistmach, and and error message, before getting to the version one. Check:

```cmake
if (NOT CMAKE_CXX_COMPILER_ID MATCHES ${EXPECTED_CMAKE_CXX_COMPILER_ID})
            message(FATAL_ERROR "Incorrect '${CONAN_COMPILER}'. Toolset specifies compiler as '${EXPECTED_CMAKE_CXX_COMPILER_ID}' "
                                "but CMake detected '${CMAKE_CXX_COMPILER_ID}'")
        endif()
```

The error message is perfect now, thanks!

Reply:
But if we don't check here again against CONAN_COMPILER, than the error happens in case of profile mismatch, as you commented previously in the other commit. I can remove that part of not required and not desired

Reply:
I am not sure if I follow. CONAN_COMPILER is checked before calling ``check_compiler_version()``, so in theory, inside ``check_compiler_version()`` it is not possible that there is a compiler mismatch. What am I missing here? There is probably something that I dont see.